### PR TITLE
UAR-1211: Statement date validations

### PR DESCRIPTION
**Pull request by jkayyala**
### JIRA link
Jira [UAR-1108](https://companieshouse.atlassian.net/browse/UAR-1108)

### Change description
Add validations so that start date of BO is before the filing date, as the filing date is not present in the BO page, filing date is retrieved from session.

Renamed checkCeasedDateOnOrAfterStartDate to checkFirstDateOnOrAfterSecondDate as its now used to check for dates other than ceasedDate and startDate

### Work checklist

- [x] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

[UAR-1108]: https://companieshouse.atlassian.net/browse/UAR-1108?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/validation/custom.validation.ts [hunk 1]
```diff
@@ -505,6 +506,23 @@ export const checkBeneficialOwnersSubmission = (req) => {
   return true;
 };
 
+export const checkFilingPeriod = (req, secondDateDay: string, secondDateMonth: string, secondDateYear: string, errorMessage: string) => {
```
Comment (AndreKramer7654) [inline src/validation/custom.validation.ts:509]: Could we not use FilingPeriod? checkPreviousToFilingDate?

Comment (AndreKramer7654) [reply src/validation/custom.validation.ts:509]: also arg should probably be just date and not secondDate

File: src/validation/fields/date.validation.ts [hunk 2]
```diff
@@ -17,7 +17,8 @@ import {
   MonthFieldErrors,
   YearFieldErrors,
   checkDateIPLegalEntityBO,
-  checkCeasedDateOnOrAfterStartDate
```
Comment (AndreKramer7654) [inline src/validation/fields/date.validation.ts:20]: we could just define checkCeasedDateOnOrAfterStartDate in terms of checkFirstDateOnOrAfterSecondDate and leave callers as is. Might be clearer too.

**Review by bantao-ch — APPROVED** [review]

File: src/validation/fields/date.validation.ts [hunk 3]
```diff
@@ -67,19 +69,30 @@ const is_still_active_validations = (date_field_id: string, radio_button_id: str
 
 const is_trust_still_active_validation = (error_message: string) => [
   body("endDate")
-    .custom((value, { req }) => checkCeasedDateOnOrAfterStartDate(
```
Comment (AndreKramer7654) [inline src/validation/fields/date.validation.ts:70]: revert this change?

Comment (AndreKramer7654) [reply src/validation/fields/date.validation.ts:70]: ok new name is better!

**Review by AndreKramer7654 — APPROVED** [review]
